src/ooxml/java/org/apache/poi/xssf/usermodel/XSSFTableColumn.java

```diff
@@ -0,0 +1,109 @@
+/* ====================================================================
+   Licensed to the Apache Software Foundation (ASF) under one or more
+   contributor license agreements.  See the NOTICE file distributed with
+   this work for additional information regarding copyright ownership.
+   The ASF licenses this file to You under the Apache License, Version 2.0
+   (the "License"); you may not use this file except in compliance with
+   the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+==================================================================== */
+
+package org.apache.poi.xssf.usermodel;
+
+import org.apache.poi.xssf.usermodel.helpers.XSSFXmlColumnPr;
+import org.openxmlformats.schemas.spreadsheetml.x2006.main.CTTableColumn;
+import org.openxmlformats.schemas.spreadsheetml.x2006.main.CTXmlColumnPr;
+
+/**
+ * A column of a {@link XSSFTable}.
+ * 
+ * @author Leonard Kappe
+ * @since 4.0.0
+ */
+public class XSSFTableColumn {
+
+    private final XSSFTable table;
+    private final CTTableColumn ctTableColumn;
+    private XSSFXmlColumnPr xmlColumnPr;
+
+    /**
+     * Create a new table column.
+     *
+     * @param table
+     *            the table which contains the column
+     * @param ctTableColumn
+     *            the table column xmlbean to wrap
+     * @since 4.0.0
+     */
+    public XSSFTableColumn(XSSFTable table, CTTableColumn ctTableColumn) {
```

Comment:
I would insist in having all methods that use a `CT*` parameter to be annotated as `@Internal` when they are not private.

Reply:
I'll check the code and add those annotations.

This particular constructor doesn't seem to be useful for the user anyway. I think I could make this constructor's visibility be `protected` instead of `public` and then cleanup `XSSFTable.addColumn` and make it part of the public API. `XSSFTable.addColumn` then is the correct and only way to create new table columns in the API.

Reply:
Definitively the best option!